### DISCUSSION TITLE
permission denied github push 403

### DISCUSSION THREAD

**Discussion by Ufik0203** [community/community#87141] [Programming Help] [answered]
### Select Topic Area

Question

### Body

Hello guys, i have trouble when i wanna commit using git. the trouble notification is "Permission denied" and 403 code. How to solve this?

**Comment by Awazonn** [comment]
verify it or uninstall then reinstall it

**Reply by Awazonn** [reply]
brt

**Comment by faridnec** [comment, marked as answer]
Hey, you're probably facing permission issues while trying to commit in Git (likely indicates an issue with your authentication). Check your SSH keys or credentials, and ensure you have the necessary permissions on the remote repository. If using HTTPS, make sure your username and password are correct. If the problem persists, double-check repository settings and contact your Git hosting provider for detail.

**Reply by Ufik0203** [reply]
it work , thank you

**Comment by pavanpsai** [comment]
Just Go to user credentials manager in windows and delete the (to) credentials. This what I done for this issue. Hope this resolve the issue. Thanks :)

![Screenshot 2024-10-13 194040](https://github.com/user-attachments/assets/2789a804-78c2-458c-97c9-bb3613440f2d)

**Reply by Manzi004** [reply]
this solution works perfectly
had the same issue in the past, this resolved the issue 

happy coding :)

**Reply by lazysegtree** [reply]
This worked. Thanks @pavanpsai

**Reply by a1029563229** [reply]
Thanks a lot! This resolved my problem.

**Reply by Mackenzie-Mealy-2025** [reply]
super helpful was trapped with a DE course creds had no clue how to get free. thanks a ton

**Reply by livialop** [reply]
worked for me!! thank you.

**Reply by jjhuang417** [reply]
Worked amazingly! Thank you so much

**Comment by CristyNel** [comment]
To solve the problem of the error "Permission denied (publickey)" when trying to push to a GitHub repository, you need to add your SSH key to the ssh-agent and configure Git to use it.

```sh
GitSshKey=<path to your ssh key>
GitUsername=<your git username>
GitEmail="<your git email>"
chmod 600 $GitSshKey
ssh-add $GitSshKey
git config --global user.name $GitUsername # Set GitHub username
git config --global user.email $GitEmail # Set GitHub email
git config --global http.sslBackend "openssl"
```
